pkgs/stdenv/generic/check-meta.nix

```diff
@@ -126,7 +126,7 @@ let
       '';
 
   throwEvalHelp = { reason , errormsg ? "" }:
-    throw (''
+    (if reason != "unknown-meta" then throw else (x : builtins.trace x true)) (''
```

Comment:
I'd be tempted to make this configurable so we can highlight or ignore certain conditions depending on what we're testing for:

```diff
diff --git a/pkgs/stdenv/generic/check-meta.nix b/pkgs/stdenv/generic/check-meta.nix
index 03a85633ed6..308a0f370c7 100644
--- a/pkgs/stdenv/generic/check-meta.nix
+++ b/pkgs/stdenv/generic/check-meta.nix
@@ -125,11 +125,12 @@ let
 
       '';
 
-  throwEvalHelp = { reason , errormsg ? "" }:
+  handleEvalIssueDefault = { reason , errormsg ? "" }:
     throw (''
       Package ‘${attrs.name or "«name-missing»"}’ in ${pos_str} ${errormsg}, refusing to evaluate.
 
       '' + ((builtins.getAttr reason remediation) attrs));
+  handleEvalIssue = x: (config.handleEvalIssue or handleEvalIssueDefault) x;
 
   metaTypes = with lib.types; rec {
     # These keys are documented
@@ -192,7 +193,7 @@ let
   validityCondition =
          let v = checkValidity attrs;
          in if !v.valid
-           then throwEvalHelp (removeAttrs v ["valid"])
+           then handleEvalIssue (removeAttrs v ["valid"])
            else true;
 
 in
```

Reply:
If we play our cards right, we could take `evaluation aborted with the following error message: ‘Package ‘aangifte2006-1’ in /home/grahamc/projects/nixpkgs/pkgs/applications/taxes/aangifte-2006/default.nix:15 has an invalid meta attrset:
` parse the path an line, and leave in-line feedback on PRs..........

Reply:
I'm also tempted by this.

Reply:
Oh, I think this will be a good way to clean https://hydra.nixos.org/jobset/nixpkgs/trunk#tabs-errors 👍